tests/api/crud/test_secrets.py

```diff
@@ -31,6 +35,415 @@ def test_store_secrets_verifications(
+        project, {valid_secret_key: valid_secret_value}
+    )
+
+    # store secret with invalid key - map should be used
+    mlrun.api.crud.Secrets().store_secrets(
+        project,
+        mlrun.api.schemas.SecretsData(
+            provider=provider, secrets={invalid_secret_key: invalid_secret_value}
+        ),
+        True,
+        key_map_secret_key,
+    )
+    k8s_secrets_mock.assert_project_secrets(
+        project,
+        {
+            valid_secret_key: valid_secret_value,
+            0: invalid_secret_value,
+            key_map_secret_key: json.dumps({invalid_secret_key: 0}),
+        },
+    )
+
+    # store secret with the same invalid key and different value
+    mlrun.api.crud.Secrets().store_secrets(
+        project,
+        mlrun.api.schemas.SecretsData(
+            provider=provider, secrets={invalid_secret_key: invalid_secret_value_2}
+        ),
+        True,
+        key_map_secret_key,
+    )
+    k8s_secrets_mock.assert_project_secrets(
+        project,
+        {
+            valid_secret_key: valid_secret_value,
+            0: invalid_secret_value_2,
+            key_map_secret_key: json.dumps({invalid_secret_key: 0}),
+        },
+    )
+
+    # store secret with the different invalid key and value - do it twice - nothing should change
+    for _ in range(2):
+        mlrun.api.crud.Secrets().store_secrets(
+            project,
+            mlrun.api.schemas.SecretsData(
+                provider=provider,
+                secrets={invalid_secret_2_key: invalid_secret_2_value},
+            ),
+            True,
+            key_map_secret_key,
+        )
+        k8s_secrets_mock.assert_project_secrets(
+            project,
+            {
+                valid_secret_key: valid_secret_value,
+                0: invalid_secret_value_2,
+                1: invalid_secret_2_value,
+                key_map_secret_key: json.dumps(
+                    {invalid_secret_key: 0, invalid_secret_2_key: 1}
+                ),
+            },
+        )
+
+    # change values to all secrets
+    mlrun.api.crud.Secrets().store_secrets(
+        project,
+        mlrun.api.schemas.SecretsData(
+            provider=provider,
+            secrets={
+                valid_secret_key: valid_secret_value_2,
+                invalid_secret_key: invalid_secret_value,
+                invalid_secret_2_key: invalid_secret_2_value_2,
+            },
+        ),
+        True,
+        key_map_secret_key,
+    )
+    k8s_secrets_mock.assert_project_secrets(
+        project,
+        {
+            valid_secret_key: valid_secret_value_2,
+            0: invalid_secret_value,
+            1: invalid_secret_2_value_2,
+            key_map_secret_key: json.dumps(
+                {invalid_secret_key: 0, invalid_secret_2_key: 1}
+            ),
+        },
+    )
+
+
+def _mock_secrets_crud_uuid_generation():
+    uuids_list = [i for i in range(10000)]
+    uuids_iter = iter(uuids_list)
```

Comment:
Not that it matter too much as this is a test, but why not simply hold a single `i=0` and on `_mock_generate_uuid()` increment and return? You're holding a 10000 items range in memory for no good reason...

Reply:
In order to be able to "remember" the current number between different calls to `_mock_generate_uuid` note that I'm using a variable that is defined outside of the scope of the function.
primitive types (like the integer you suggest) can't be used for that purpose since they are passed by value unlike list or iter that are passed by reference

Reply:
I changed it to be `iter(range(10000))` that way the list isn't really created and numbers are created on demand